pkg/kcp/provider/azure/meta/metadata.go

```diff
@@ -100,6 +103,9 @@ func GetErrorMessage(err error) (string, bool) {
 			return VnetAddressSpaceOverlapsWithAlreadyPeeredVnetMessage, true
 		case InvalidResourceName:
 			return InvalidResourceNameMessage, true
+		case VnetAddressSpacesOverlap:
+			r := regexp.MustCompile(`Overlapping address prefixes:[^\"]*`)
```

Comment:
what happens if this doesnt compile? will it panic and crash the app?

Reply:
There is no user input in regex therefore we are responsible to compile. Yes, it will panic if the expression does not compile.

Reply:
So if azure changes their message, our app will panic? 
If so, this must be implemented in a way to avoid potential panic.